test/realm_test.dart

```diff
@@ -1150,4 +1150,163 @@ Future<void> main([List<String>? args]) async {
       expect(read, [person]);
     });
   });
+  group('IsValid tests:', () {
+    test('RealmObject isValid only if realm is opened', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      var team = Team("team one");
+      expect(team.isValid, true);
+      realm.write(() {
+        realm.add(team);
+      });
+      expect(team.isValid, true);
+      realm.close();
+      expect(team.isValid, false);
+    });
+
+    test('List isValid only if realm is opened', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      realm.write(() {
+        realm.add(Team("Speed Team", players: [
+          Person("Michael Schumacher"),
+          Person("Sebastian Vettel"),
+          Person("Kimi Räikkönen"),
+        ]));
+      });
+
+      var teams = realm.all<Team>();
+
+      expect(teams, isNotNull);
+      expect(teams.length, 1);
+      RealmList<Person> teampPlayers = teams[0].players as RealmList<Person>;
```

Comment:
As we discussed I suggest we fix the generator first before we merge isValid. So we will not need to do the casts here